go-controller/pkg/ovn/ovn.go

```diff
@@ -780,12 +780,29 @@ func (oc *Controller) WatchNetworkPolicy() {
 		AddFunc: func(obj interface{}) {
 			policy := obj.(*kapisnetworking.NetworkPolicy)
 			oc.initRetryPolicy(policy)
+			oc.checkAndSkipRetryPolicy(policy)
+			// If there is a delete entry this is a network policy being added
+			// with the same name as a previous network policy that failed deletion.
+			// Destroy it first before we add the new policy.
+			if oc.retryEntryHasDelete(policy) {
```

Comment:
you could just return the `entry.oldPolicy` back from `retryEntryHasDelete` function, based on whether its nil or not you can use that value for triggering the delete. I am ok with calling the `getRetryEntry` again like you are doing now as well just thinking out loud.

Reply:
Yeah, I agree; let's just have retryEntryHasDelete -> `if oldPolicy := retryEntryGetDelete(); oldPolicy != nil {`

Reply:
done